### ISSUE TITLE
Cannot read property 'getContext' of undefined

### ISSUE THREAD

**Issue by phanchanra** [apertureless/vue-chartjs#631]
### Expected Behavior
import { Bar } from 'vue-chartjs'
Bar chart color with gradient
`this.gradient = this.$refs.canvas
              .getContext('2d')
              .createLinearGradient(0, 0, 0, 450)
            this.gradient.addColorStop(0, 'rgba(255, 0,0, 0.5)')
            this.gradient.addColorStop(0.5, 'rgba(255, 0, 0, 0.25)')
            this.gradient.addColorStop(1, 'rgba(255, 0, 0, 0)')`
### Actual Behavior
But it is error 

- Cannot read property 'getContext' of undefined

### Environment
- vue.js version: 2.6.11
- vue-chart.js version: 3.5.0
- npm version: 6.14.4

**Comment by apertureless** [comment]
Keep in mind, that if you are using Single File Components, you need to remove the `<template>` tag, because it can't be merged with the extended component.

https://vue-chartjs.org/guide/#vue-single-file-components

**Comment by phanchanra** [comment]
I remove template already but it still show above the same error

**Comment by blakeja** [comment]
@phanchanra 

With typescript, cast ```this.$refs.canvas``` to ```<HTMLCanvasElement>```

```
const canvas = <HTMLCanvasElement>this.$refs.canvas;

const gradient = canvas.getContext('2d')
  .createLinearGradient(0, 0, 0, 450);

gradient.addColorStop(0, 'rgba(255, 0,0, 0.9)')
gradient.addColorStop(0.5, 'rgba(255, 0, 0, 0.5)');
gradient.addColorStop(1, 'rgba(255, 0, 0, 0)');
```

**Comment by blakeja** [comment]
@phanchanra 

You will need to cast ```this.$refs.canvas``` as follows:

```
const canvas = this.$refs.canvas as HTMLCanvasElement;

const gradient = canvas.getContext('2d')
  .createLinearGradient(0, 0, 0, 450);

gradient.addColorStop(0, 'rgba(255, 0,0, 0.9)')
gradient.addColorStop(0.5, 'rgba(255, 0, 0, 0.5)');
gradient.addColorStop(1, 'rgba(255, 0, 0, 0)');
```

**Comment by phanchanra** [comment]
thank for reply now it work fine

**Comment by arishojaei** [comment]
> Keep in mind, that if you are using Single File Components, you need to remove the `<template>` tag, because it can't be merged with the extended component.
> 
> https://vue-chartjs.org/guide/#vue-single-file-components
> 
> Otherwise, please post a reproduction and not only 2 lines of code.

The link you provide doesn't even exist!

**Comment by apertureless** [comment]
@arishojaei 

Yeah because this issue is 2 years old. And vue-chartjs and the documentation also got reworked with the vue3 support.
